File: exercises/small_problems/easy5/letter_counter2.rb
def word_sizes(str)
  size_hash = {}

  str_arr = str.split.map! do |word| 
    word.each_char do |char|
      word.delete!(char) unless char =~ /[a-zA-Z]/
    end

    word.size 
  end

  str_arr.each do |num|
    size_hash.include?(num) ? size_hash[num] += 1 : size_hash[num] = 1
  end

  size_hash
end

p word_sizes('Four score and seven.') == { 3 => 1, 4 => 1, 5 => 2 }
p word_sizes('Hey diddle diddle, the cat and the fiddle!') == { 3 => 5, 6 => 3 }
p word_sizes("What's up doc?") == { 5 => 1, 2 => 1, 3 => 1 }
p word_sizes('')== {}

File: exercises/jsSmallProblems/easy3/letterCounter2.js
const wordSizes = (string) => {
  const result = {};
  if (string.trim().length === 0) return {};

  const lengths = string.split(' ').map((word) => word.replace(/[^a-z]/gi, '').length);

  for (let i = 0; i < lengths.length; i += 1) {
    result[lengths[i]] = result[lengths[i]] || 0;
    result[lengths[i]] += 1;
  }

  return result;
};

console.log(wordSizes('Four score and seven.')); // { "3": 1, "4": 1, "5": 2 }
console.log(wordSizes('Hey diddle diddle, the cat and the fiddle!')); // { "3": 5, "6": 3 }
console.log(wordSizes("What's up doc?")); // { "5": 1, "2": 1, "3": 1 }
console.log(wordSizes('')); // {}
